front/src/tabletview/rangeofficer.js

```diff
@@ -125,21 +132,46 @@ const TrackButtons = ({track, tracks, setTracks, scheduleId, tablet, fin}) => {
     text = tablet.Red[fin];
   }
 
+  const [textState, setTextState] = useState(text);
+  socket.on('trackUpdate', msg => {
+    if(msg.id === track.id){
```

Comment:
Technically this could be done more efficiently (processing power-wise) with a switch-case, although it isn't necessary by any means.